core/src/main/java/org/mule/routing/filters/ExpressionFilter.java

```diff
@@ -53,6 +56,7 @@
     private ExpressionConfig config;
     private String fullExpression;
     private boolean nullReturnsTrue = false;
+    private boolean nonBooleanReturnsTrue = parseBoolean(getProperty(MULE_EXPRESSION_FILTER_DEFAULT_BOOLEAN_VALUE, "true"));
```

Comment:
Remember to remove this from the 3.x solution, you can directly modify the `expression-filter` XSD to allow the new attribute there (you'll need a functional test there to validate it works).

Reply:
You can remove all System property related code and leave it for 3.9.x.